keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -19,7 +19,18 @@ sealed class DeliveryArtifact {
   abstract val versioningStrategy: VersioningStrategy
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
+
   override fun toString() = "$type:$name" + if (reference != null) " (ref: $reference)" else ""
+
+  override fun equals(other: Any?): Boolean {
+    if (other is DeliveryArtifact) {
+      return other.name == this.name &&
+        other.type == this.type &&
+        other.reference == this.reference &&
+        other.deliveryConfigName == this.deliveryConfigName
```

Comment:
This now compares only the fields that actually identify the object to determine if they're "equal". Note that I'm not overriding `hashCode()` here, so objects with differences in other properties would still be considered different keys if you add them to a map, for example.